Shop/Shop.DAL/Repositories/GenericRepository.cs

```diff
@@ -16,7 +16,7 @@ public GenericRepository(ApplicationContext applicationContext)
             DbSet = applicationContext.Set<TEntity>();
         }
 
-        public async Task<TEntity> Create(TEntity item, CancellationToken ct)
+        public virtual async Task<TEntity> Create(TEntity item, CancellationToken ct)
         {
             await DbSet.AddAsync(item, ct);
```

Comment:
property should be called as CamelCase and fields should be called as pascalCase